src/core/EdFi.DataManagementService.Core/Validation/EqualityConstraintValidator.cs

```diff
@@ -33,15 +41,36 @@ public string[] Validate(JsonNode? documentBody, IEnumerable<EqualityConstraint>
             var sourcePathResult = sourcePath.Evaluate(documentBody);
             var targetPathResult = targetPath.Evaluate(documentBody);
 
-            Trace.Assert(sourcePathResult.Matches != null, "Evaluation of sourcePathResult.Matches resulted in unexpected null");
-            Trace.Assert(targetPathResult.Matches != null, "Evaluation of targetPathResult.Matches resulted in unexpected null");
+            Trace.Assert(
+                sourcePathResult.Matches != null,
+                "Evaluation of sourcePathResult.Matches resulted in unexpected null"
+            );
+            Trace.Assert(
+                targetPathResult.Matches != null,
+                "Evaluation of targetPathResult.Matches resulted in unexpected null"
+            );
 
             var sourceValues = sourcePathResult.Matches.Select(s => s.Value);
             var targetValues = targetPathResult.Matches.Select(t => t.Value);
 
             if (!AllEqual(sourceValues.Concat(targetValues).ToList()))
             {
-                errors.Add($"Constraint failure: document paths {equalityConstraint.SourceJsonPath.Value} and {equalityConstraint.TargetJsonPath.Value} must have the same values");
+                string conflictValues = string.Join(
+                    ", ",
+                    sourceValues
+                        .Concat(targetValues)
+                        .Distinct(new JsonNodeEqualityComparer())
+                        .Select(x => $"'{x}'")
+                        .ToArray()
+                );
+                var targetSegment = targetPath.Segments[1].ToString().Replace(".", "");
```

Comment:
I don't know that this will always be the proper way to get the property name you're looking for. It looks like you're assuming a path like `{$.schoolReference.schoolId}` and extracting `schoolId`. Instead of hard coding the segment at index 1, it might be safer to take the segment at the _last_ index. Also instead of `.Replace` I would suggest either `TrimStart` or `[1..]`